app/src/main/java/com/allan/boardbuddies/activities/CanvasActivity.java

```diff
@@ -140,17 +121,37 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
     }
 
     private void showDialog(String resultKey) {
-        // DialogFragment.show() will take care of adding the fragment
-        // in a transaction.  We also want to remove any currently showing
-        // dialog, so make our own transaction and take care of that here.
         FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
         Fragment prev = getSupportFragmentManager().findFragmentByTag("dialog");
         if (prev != null) {
             ft.remove(prev);
         }
         ft.addToBackStack(null);
-
         DialogFragment newFragment = TextDialogFragment.newInstance("", resultKey);
         newFragment.show(ft, "dialog");
     }
+
+    private String saveTextNote() {
```

Comment:
nit: I really feel like you should start making a FileUtil helper class by now, since you're dealing with a lot of reads and writes